onboarding/source/amb_light_service.c

```diff
@@ -26,14 +59,58 @@ uint8_t initLightService(void) {
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+
+    light_event_t event;
+    while(1) {
+        if (xQueueReceive(xQueue1, &event, (TickType_t) 0) == pdPASS) {
+            if(event == MEASURE_LIGHT) {
+                // if event recieved is MEASURE_LIGHT, get value from adc and send over serial
+                uint8_t data = getAmbientLightData();
+                sciPrintText(scilinREG, &data, sizeof(uint8_t));
+            }
+        }
+    }
 
     /* USER CODE END */
 }
 
 uint8_t sendToLightServiceQueue(light_event_t *event) {
     /* USER CODE BEGIN */
     // Send the event to the queue.
-    
+    if (xQueueSend(xQueue1, (void *) &event, (TickType_t) 0) == pdPASS) {
```

Comment:
xQueueSend expects a pointer to the event. Since `sendToLightServiceQueue` takes this pointer as an argument, you can just pass `(void *) event` into `xQueueSend`